core-bundle/src/Messenger/Transport/AutoFallbackTransportFactory.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Messenger\Transport;
+
+use Contao\CoreBundle\Messenger\AutoFallbackNotifier;
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Messenger\Exception\InvalidArgumentException;
+use Symfony\Component\Messenger\Transport\Serialization\SerializerInterface;
+use Symfony\Component\Messenger\Transport\TransportFactoryInterface;
+use Symfony\Component\Messenger\Transport\TransportInterface;
+
+class AutoFallbackTransportFactory implements TransportFactoryInterface
+{
+    public function __construct(private AutoFallbackNotifier $autoFallbackNotifier, private ContainerInterface $messengerTransportLocator)
+    {
+    }
+
+    public function createTransport(string $dsn, array $options, SerializerInterface $serializer): TransportInterface
+    {
+        if (false === $parsedUrl = parse_url($dsn)) {
+            throw new InvalidArgumentException(sprintf('The given Auto Fallback DSN "%s" is invalid.', $dsn));
+        }
+
+        $target = isset($parsedUrl['path']) ? substr($parsedUrl['path'], \strlen('contao_auto_fallback://')) : '';
```

Comment:
The underscore is an invalid scheme character, because of this the `path` does not strip the protocol part here automatically.

I think we have to change the scheme name to `contao-auto-fallback` to make it a valid DSN. _(The Symfony example also uses a dash, like `my-transport://`)_

```suggestion
        $target = $parsedUrl['path'] ?? '';
```

Reply:
Ah right, it's `host` though, not `path` then ;) I'll fix it.